sfepy/discrete/common/extmods/cmesh.pyx

```diff
@@ -688,7 +688,12 @@ cdef class CMesh:
         ccf = self.get_conn(td, td - 1)
 
         out = np.empty((ccf.n_incident, self.dim), dtype=np.float64)
-        mesh_get_facet_normals(self.mesh, &out[0, 0], which)
+        if self.dim == 1:
+            # TODO quick fix for normals, relies on nice ordering of 1D mesh
```

Comment:
Is it still TODO? :) See also fb3c3d476e10562a821a7c0594fd87080c35716a - something similar was done for SurfaceMapping.

Reply:
I have checked fb3c3d4 however it does not work for me.  I'm not using SurfaceMapping in DG only VolumeMapping so maybe I missed something, but it does not return normals for each face but for each qp, which is not what I need. Also wouldn't it be nice if cmesh returned correct normals even for 1D?

Reply:
Yes, I am aware that it does not help your case - it was just to show that a similar "hack" (special case overwriting in Python) had been used before.

In 1D, the surface is a single point, so the C function cannot guess the correct direction - the information is missing. So, yes, it would be nice, but it is not worth the effort with the current C code/data structures.